src/OrchardCore.Modules/OrchardCore.Localization/Views/CulturePickerContainer.cshtml

```diff
@@ -0,0 +1,22 @@
+@using OrchardCore.Localization
+
+@{
+    var selectedCultureName = Model.CurrentCulture.Name;
+}
+<select id="culture" name="culture" class="form-select me-3" onchange="changeCulture()">
+  @foreach (var culture in Model.SupportedCultures)
+  {
+    <option selected="@(culture.Name == selectedCultureName ? "selected" : null)" value="@culture.Name">@culture.DisplayName</option>
+  }
+</select>
+
+<script at="Foot">
+    function changeCulture() {
+        var culture = document.getElementById('culture');
+        if (culture != '@selectedCultureName') {
+            var cookieValue = '@CulturePickerStartup.AdminSiteCookieName=c=' + culture.options[culture.selectedIndex].value + '|uic=' + culture.options[culture.selectedIndex].value;
+            document.cookie = cookieValue;
+            window.location = window.location.href.split('?')[0];
```

Comment:
Maybe use jquery, on an inline script we can even use `depends-on="jquery"`.

And maybe use the jQuery change() event.

Normally to pass data from a view to a script we use html data attributes that the script can then retrieve.

Reply:
May be, now the important thing to me is provider a proper way to show the picker when the feature is on, then I can consider your suggestion here

Reply:
> Normally to pass data from a view to a script we use html data attributes that the script can then retrieve.

What does this comment refer to?

Reply:
> What does this comment refer to?

Here related to the above script using directly e.g. `@selectedCultureName` with the razor `@` operator.

So I suggested to retrieve data from html attributes as we do in many other places.

Reply:
I think it's not a problem while it's already rendered on the server, anyhow I will update this soon

Reply:
@jtkech hope the latest change is what you meant before?